detection_rules/custom_schemas.py

```diff
@@ -0,0 +1,44 @@
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0; you may not use this file except in compliance with the Elastic License
+# 2.0.
+
+"""Custom Schemas management."""
+from pathlib import Path
+
+import eql
+import eql.types
+
+from .config import parse_rules_config
+from .utils import cached
+
+RULES_CONFIG = parse_rules_config()
+
+
+@cached
+def get_custom_schemas(stack_version: str) -> dict:
+    """Load custom schemas if present."""
+    custom_schema_dump = {}
+    stack_schema_map = RULES_CONFIG.stack_schema_map[stack_version]
+
+    for schema, value in stack_schema_map.items():
+        if schema not in ["beats", "ecs", "endgame"]:
```

Comment:
what happens if it _is_ one of these three?

Reply:
I think it should ignore it if it is one of those, as those are reserved "schema words" that are by definition not custom. Do you agree?

Reply:
yes I think that is the smart approach, but wont this return an empty dict in this case? It would be better to raise an error, forbidding using reserved words. Thoughts?

Reply:
I think that is a good idea, we will need to have it function such that it will allow the words once, as we would want them to be able to specify beats, etc. or to not use those if desired. But we would not want them to use multiple as the result would be confusing :+1:

Reply:
Upon further testing and reflection, I think the test for additional beats, ecs, or endgame schemas should not be in the `custom_schemas.py` as it might be confusing to have the custom schema loader be validating non-custom schemas. I think this check would be considered increased validation for the schema map in general. As such, I think this could target main, as the check is not DAC specific.